server/protocols/webadmin/webadmin-mailbox/src/main/java/org/apache/james/webadmin/service/CreateMissingParentsTask.java

```diff
@@ -0,0 +1,160 @@
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.model.MailboxPath;
+import org.apache.james.server.task.json.dto.TaskDTO;
+import org.apache.james.server.task.json.dto.TaskDTOModule;
+import org.apache.james.task.Task;
+import org.apache.james.task.TaskExecutionDetails;
+import org.apache.james.task.TaskType;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.github.fge.lambdas.Throwing;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+
+public class CreateMissingParentsTask implements Task {
+    static final TaskType TASK_TYPE = TaskType.of("CreateMissingParentsTask");
+
+    public static class AdditionalInformation implements TaskExecutionDetails.AdditionalInformation {
+        private static AdditionalInformation from(Username username, long totalCreated) {
+            return new AdditionalInformation(Clock.systemUTC().instant(), username, totalCreated);
+        }
+
+        private final Username username;
+        private final Instant timestamp;
+        private final long totalCreated;
+
+        public AdditionalInformation(Instant timestamp, Username username, long totalCreated) {
+            this.username = username;
+            this.totalCreated = totalCreated;
+            this.timestamp = timestamp;
+        }
+
+        public String getUsername() {
+            return username.asString();
+        }
+
+        public long getTotalCreated() {
+            return totalCreated;
+        }
+
+        @Override
+        public Instant timestamp() {
+            return timestamp;
+        }
+    }
+
+    public static class CreateMissingParentsTaskDTO implements TaskDTO {
+        private final String type;
+        private final String username;
+
+        public CreateMissingParentsTaskDTO(@JsonProperty("type") String type,
+                                           @JsonProperty("username") String username) {
+            this.type = type;
+            this.username = username;
+        }
+
+        @Override
+        public String getType() {
+            return type;
+        }
+
+        public String getUsername() {
+            return username;
+        }
+    }
+
+    public static TaskDTOModule<CreateMissingParentsTask, CreateMissingParentsTask.CreateMissingParentsTaskDTO> module(MailboxManager mailboxManager) {
+        return DTOModule
+            .forDomainObject(CreateMissingParentsTask.class)
+            .convertToDTO(CreateMissingParentsTask.CreateMissingParentsTaskDTO.class)
+            .toDomainObjectConverter(dto -> new CreateMissingParentsTask(mailboxManager, Username.of(dto.username)))
+            .toDTOConverter((task, type) -> new CreateMissingParentsTask.CreateMissingParentsTaskDTO(type, task.username.asString()))
+            .typeName(TASK_TYPE.asString())
+            .withFactory(TaskDTOModule::new);
+    }
+
+    private final Username username;
```

Comment:
I was thinking this task creates the missing parents for **all** mailboxes and is thus not bound to a user.

Reply:
then shouldn't we have to use mailboxDAO instead of mailboxManager?

Reply:
We can use a static username here.

No we are outside of the mailbox project we should use `MailboxManager`.